src/services/download/gcs.rs

```diff
@@ -273,3 +273,153 @@ impl GcsDownloader {
         .collect()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::sources::{CommonSourceConfig, DirectoryLayoutType, SourceId};
+    use crate::test;
+    use crate::types::ObjectType;
+
+    use super::*;
+    use sha1::{Digest as _, Sha1};
+
+    fn gcs_source_key() -> Option<GcsSourceKey> {
+        let private_key = std::env::var("SENTRY_SYMBOLICATOR_GCS_PRIVATE_KEY").ok()?;
+        let client_email = std::env::var("SENTRY_SYMBOLICATOR_GCS_CLIENT_EMAIL").ok()?;
+
+        if private_key.is_empty() || client_email.is_empty() {
+            None
+        } else {
+            Some(GcsSourceKey {
+                private_key,
+                client_email,
+            })
+        }
+    }
+
+    macro_rules! gcs_source_key {
```

Comment:
interesting that you can have name conflicits between functions and macros…
Maybe its worth inlining?